logdb/logdb.go

```diff
@@ -398,11 +398,24 @@ func (db *LogDB) NewWriterSyncOff() *Writer {
 
 func topicValue(topics []thor.Bytes32, i int) []byte {
 	if i < len(topics) {
-		return topics[i][:]
+		return removeLeadingZeros(topics[i][:])
 	}
 	return nil
 }
 
+func removeLeadingZeros(bytes []byte) []byte {
+	i := 0
+	// increase i until it reaches the first non-zero byte
+	for ; i < len(bytes) && bytes[i] == 0; i++ {
+	}
+	bytes = bytes[i:]
+	// ensure at least 1 byte
+	if len(bytes) == 0 {
```

Comment:
Any particular reason for this?

Reply:
NVM, I got that.

Reply:
Pushed this change a change for:
```golang
if i == len(bytes) {
	return []byte{0}
}
```
because the unit test was failing when you pass in a number of bytes not equal to 32